play/play-v28/RotatingSecretComponents.scala

```diff
@@ -59,4 +66,39 @@ object RotatingSecretComponents {
     val snapshotProvider: SnapshotProvider) extends FlashCookieBaker with RotatingSecretCookieCodec {
     override val jwtConfiguration: JWTConfiguration = config.jwt
   }
+
+  class RotatingKeyCSRFTokenSigner(
```

Comment:
This class is the alternative implementation to Play's [`DefaultCSRFTokenSigner`](https://github.com/playframework/playframework/blob/1197129f84390c45a7d013832a527f3d5d4a505c/core/play/src/main/scala/play/api/libs/crypto/CSRFTokenSigner.scala#L76) - our alternative implementation can cope with rotating keys.

Our implementation borrows some code as copy'n'paste from `DefaultCSRFTokenSigner` where it's unavoidable - specifically `compareSignedTokens()` - but delegates to underlying instances of `DefaultCSRFTokenSigner` as much as possible. If you look at the implementation of `DefaultCSRFTokenSigner` you can see it's doing some [relatively obscure crypto things](https://github.com/playframework/playframework/blob/1197129f84390c45a7d013832a527f3d5d4a505c/core/play/src/main/scala/play/api/libs/crypto/CSRFTokenSigner.scala#L77-L82) that we'd prefer delegate, rather than chase & re-implement ourselves.